### PR TITLE
obs-studio: Allow package to be configurable

### PR DESCRIPTION
**Pull request by starcraft66**
### Description

This PR adds an option to specify which obs package to install.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/obs-studio.nix [hunk 1]
```diff
@@ -39,7 +48,7 @@ in {
   };
 
   config = mkIf cfg.enable {
-    home.packages = [ package ];
```
Comment (kalbasit) [inline modules/programs/obs-studio.nix:42]: Can you get rid of the package variable in favor of cfg.package then?

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: modules/programs/obs-studio.nix (hunk 1 above)
Comment (starcraft66) [reply modules/programs/obs-studio.nix:42]: done

File: modules/programs/obs-studio.nix [hunk 2]
```diff
@@ -29,6 +28,15 @@ in {
     programs.obs-studio = {
       enable = mkEnableOption "obs-studio";
 
+      package = mkOption {
+        type = types.package;
+        default = pkgs.obs-studio;
+        example = literalExample "pkgs.obs-studio";
```
Comment (berbiche) [inline modules/programs/obs-studio.nix:34]:
```suggestion
        default = pkgs.obs-studio;
        defaultText = literalExample "pkgs.obs-studio";
```

**Review by berbiche — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**